File: webhook/python/webhook.py
from flask import Flask, request, jsonify
from datetime import datetime
import json, os

app = Flask(__name__)

@app.route('/webhook', methods=['POST'])
def webhook():
    try:
        data = request.get_json()
        summary = {
            "received_at": datetime.utcnow().isoformat() + "Z",
            "assistant": data.get("assistant", "Realflow Inbound Agent"),
            "brokerage": data.get("brokerage_name", "Acme Brokerage"),
            "caller_name": data.get("caller_name"),
            "caller_role": data.get("caller_role"),
            "asset_description": data.get("asset_description"),
            "asset_location": data.get("asset_location"),
            "urgency": data.get("urgency"),
            "phone": data.get("phone"),
            "email": data.get("email"),
            "transcript_summary": data.get("transcript_summary")
        }

        with open('call_logs.jsonl', 'a') as f:
            f.write(json.dumps(summary) + '\n')

        print("Saved:", summary)
        return jsonify({"status": "ok"}), 200
    except Exception as e:
        print("Error:", e)
        return jsonify({"status": "error", "error": str(e)}), 500

if __name__ == '__main__':
    app.run(port=3000)

File: webhook/node/webhook.js
const express = require('express');
const bodyParser = require('body-parser');
const fs = require('fs').promises;
const dotenv = require('dotenv');
const { logToFile } = require('./utils/logger');
dotenv.config();

const app = express();
app.use(bodyParser.json());
const PORT = process.env.PORT || 3000;

app.post('/webhook', async (req, res) => {
  try {
    const data = req.body;

    const summary = {
      received_at: new Date().toISOString(),
      assistant: data.assistant || "Realflow Inbound Agent",
      brokerage: data.brokerage_name || "Acme Brokerage",
      caller_name: data.caller_name || null,
      caller_role: data.caller_role || null,
      asset_description: data.asset_description || null,
      asset_location: data.asset_location || null,
      urgency: data.urgency || null,
      phone: data.phone || null,
      email: data.email || null,
      transcript_summary: data.transcript_summary || null
    };

    await logToFile(summary);
    res.status(200).json({ status: "ok" });
  } catch (err) {
    console.error("Webhook error:", err);
    res.status(500).json({ status: "error", error: err.message });
  }
});

app.listen(PORT, () => console.log(`Webhook running on port ${PORT}`));

